tests/codec/ber/test_decoder.py

```diff
@@ -134,17 +133,19 @@ def testIndefModeChunked(self):
             ints2octs((35, 128, 3, 2, 0, 169, 3, 2, 1, 138, 0, 0))
         ) == ((1, 0, 1, 0, 1, 0, 0, 1, 1, 0, 0, 0, 1, 0, 1), null)
 
-    def testDefModeChunkedSubst(self):
-        assert decoder.decode(
-            ints2octs((35, 8, 3, 2, 0, 169, 3, 2, 1, 138)),
-            substrateFun=lambda a, b, c: (b, b[c:])
-        ) == (ints2octs((3, 2, 0, 169, 3, 2, 1, 138)), str2octs(''))
+    # TODO: Not clear how to deal with substrateFun in stream implementation
```

Comment:
With the new streaming interface in place, may be we should not be collecting chunks and returning them all at once...? Instead, we could decode and return ASN.1 object initialized from one immediate chunk and keep returning these ASN.1 objects for each chunk till the input exceeds? That seems reasonable if we have a stream of ASN.1 objects (e.g. some media data).

Reply:
Let's move this conversation to #176